AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts

```diff
@@ -135,20 +178,32 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
     }
 
     get judgeDisplayNameInvalid() {
-        return this.judgeDisplayName.invalid && (this.judgeDisplayName.dirty || this.judgeDisplayName.touched || this.failedSubmission);
+        return (
+            this.judgeDisplayNameFld.invalid &&
+            (this.judgeDisplayNameFld.dirty || this.judgeDisplayNameFld.touched || this.failedSubmission)
+        );
+    }
+
+    get judgeEmailInvalid() {
+        return this.judgeEmailFld.invalid && (this.judgeEmailFld.dirty || this.judgeEmailFld.touched || this.failedSubmission);
+    }
+
+    get judgePhoneInvalid() {
+        return this.judgePhoneFld.invalid && (this.judgePhoneFld.dirty || this.judgePhoneFld.touched || this.failedSubmission);
     }
 
     public addJudge(judgeId: string) {
         if (judgeId) {
-            const selectedJudge = this.availableJudges.find(j => j.email === judgeId);
+            const selectedJudge = this.availableJudges.find(j => j.display_name === judgeId);
             this.judge.first_name = selectedJudge.first_name;
             this.judge.last_name = selectedJudge.last_name;
             this.judge.email = selectedJudge.email;
+            this.selectedJudgeEmail = selectedJudge.email;
             if (!this.isJudgeDisplayNameSet()) {
                 this.judge.display_name = selectedJudge.display_name;
             }
-            this.judgeDisplayName.patchValue(this.judge.display_name);
-            const newJudge = AssignJudgeComponent.mapJudgeToModel(this.judge);
+            this.judgeDisplayNameFld.patchValue(this.judge.display_name);
```

Comment:
Recommend setValue instead of patchValue

Reply:
changed